sru/sru_functional.py

```diff
@@ -232,28 +234,41 @@ def __init__(self,
         self.rescale = rescale
         self.activation_type = 0
         self.activation = 'none'
+        self.custom_u = custom_u
         if use_tanh:
             self.activation_type = 1
             self.activation = 'tanh'
 
         # projection dimension
         self.projection_size = 0
-        if n_proj > 0 and n_proj < input_size and n_proj < self.output_size:
+        if n_proj > 0 and n_proj < self.input_size and n_proj < self.output_size:
             self.projection_size = n_proj
 
         # number of sub-matrices used in SRU
         self.num_matrices = 3
+        project_first = False
         if has_skip_term and self.input_size != self.output_size:
-            self.num_matrices = 4
+            if self.custom_u is not None:
+                # If a custom u or v  is provided, project before u and v
+                project_first = True
+            else:
+                self.num_matrices = 4
+
+        if project_first:
+            input_size = self.output_size
+            self.input_to_hidden = nn.Linear(self.input_size, self.output_size)
```

Comment:
preferably disable the bias term for this `nn.Linear`.  i verified it works without bias.